modules/backing-up-red-hat-quay.adoc

```diff
@@ -0,0 +1,320 @@
++
+[source,yaml]
+----
+  metadata.creationTimestamp
+  metadata.finalizers
+  metadata.generation
+  metadata.resourceVersion
+  metadata.uid
+----
+
+. Backup the managed keys secret:
++
+[NOTE]
+====
+If you are running a version older than Red Hat Quay 3.7.0, this step can be skipped. Some secrets are automatically generated while deploying Quay for the first time. These are stored in a secret called `<quayname>quay-registry-managed-secret-keys` in the QuayRegistry namespace.
+====
++
+[source,terminal]
+----
+$ oc get secret -n <quay-namespace> <quay-registry-name>-quay-registry-managed-secret-keys -o yaml > managed-secret-keys.yaml
+----
+
+. Edit the the resulting `managed-secret-keys.yaml` file and remove all owner references. Your `managed-secret-keys.yaml` file should look similar to the following:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: <quayname>-quay-registry-managed-secret-keys
+  namespace: <quay-namespace>
+data:
+  CONFIG_EDITOR_PW: <redacted>
+  DATABASE_SECRET_KEY: <redacted>
+  DB_ROOT_PW: <redacted>
+  DB_URI: <redacted>
+  SECRET_KEY: <redacted>
+  SECURITY_SCANNER_V4_PSK: <redacted>
+----
++
+All information under the `data` property should remain the same.
+
+. Backup the current Quay configuration. This configuration lives in a secret pointed by the `QuayRegistry` custom resource and you can find the name by running the following command:
++
+[source,terminal]
+----
+$ oc get quayregistry <quay-registry-name> -n <quay-registry-namespace> -o json |jq -r .spec.configBundleSecret
+----
+
+. After obtaining the configuration bundle name, move its content into a new file:
++
+[source,terminal]
+----
+$ oc get secret -n <quay-registry-namespace> <config-bundle-secret-name> -o yaml > config-bundle.yaml
+----
+
+. Scale down the Quay component and edit the QuayRegistry custom resource by unmanaging the `horizontalpodautoscaler` component:
++
+[source,terminal]
+----
+$ oc edit quayregistry <quay-registry-name> -n <quay-registry-namespace>
+----
++
+[source,yaml]
+----
+  apiVersion: quay.redhat.com/v1
+  kind: QuayRegistry
+  metadata:
+    name: registry
+    namespace: ns
+  spec:
+    components:
+      …
+      - kind: horizontalpodautoscaler
+        managed: false
+      - kind: quay
+        managed: true
+        overrides:
+          replicas: 0
+      …
+----
++
+[NOTE]
+====
+If you are using a version older than 3.7.0, then you must manually scale down the Operator and manually scale down the Quay deployment. For example, you can run `$ oc scale --replicas=0 deployment quay-operator.v3.6.2 -n openshift-operators` to scale down the Operator, and `$ oc scale --replicas=0 deployment QUAY_MAIN_APP_DEPLOYMENT` to scale down the main application deployment.
+====
+
+. Wait for the `registry-quay-app` pods to disappear. You can check their status by running the following command:
++
+[source,terminal]
+----
+$ oc get pods -n <quay-namespace>
+----
```

Comment:
In this document, some lines are using `<quay-registry-namespace>`, some lines are using `<quay-namespace>`.  It is better to unify the string.